### PR TITLE
Easy Beta Releasing

### LINKED ISSUE #10679: Auto-Prereleasing Strategies
Where as releasing main versions of packages is simple with Changesets, pre releasing requires a different strategy. 

## Snapshot release
* create an action to release snapshots of packages automatically whenever the changeset version packages pr is updated. 
* install these packages in ci action to ensure all is well 

## Beta Release
* decide on if a more formal beta release process that runs on CI is needed. or if snapshots will be sufficient. 

## related 

 https://github.com/celo-org/celo-labs/issues/1022

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

adds a script / workflow for using changeset's pre mode. 

### TLDR run 
`yarn beta-enter`  to initiate the process 

A "Version Packges  (beta)" PR will open. Merging it will release packages

`yarn beta-exit` when complete

#### beta-mode.sh*
This script runs `changeset `and  `git` commands to ensure the correct branch is commited to and pushed up. 

#### celo-monorepo.yaml
our tests are already configured to run for any branch starting with prerelease/**

#### release.yaml 
as well as running on master, this will automatically open a pr anytime anything is pushed to prerelease/** branch.  As such there is a prepush hook to ensure changesets is in betamode before pushing. 

### Other changes

upgrade husky and fix pre-push.js

### Tested

same as used on social-connect

### Related issues

- Fixes #10679 

### Backwards compatibility

yes

### Documentation

see Release.md

### CODE REVIEW

File: scripts/hooks/pre-push.js [hunk 1]
```diff
@@ -59,17 +59,15 @@ function getDateFromFirstCommit(fromSHA, toSHA) {
 /// MAIN
 ////////////////////////////////////////////////////////////////
 
-const [remoteName, remoteUrl] = process.env.HUSKY_GIT_PARAMS.split(' ')
-
+// must trim otherwise the name will be 'origin\n'
+const remoteName = execSync('git remote').toString().trim()
```
Comment (nicolasbrugneaux) [inline scripts/hooks/pre-push.js:63]: is this wanted? the remote name came from husky before

**Review by nicolasbrugneaux — COMMENTED** [review]
Looks mostly good, just wanted to double check on the husky script

File: scripts/hooks/pre-push.js (hunk 1 above)
Comment (aaronmgdr) [reply scripts/hooks/pre-push.js:63]: yes.

Comment (aaronmgdr) [reply scripts/hooks/pre-push.js:63]: i guess that on my for not saying why i made this change. HUSKY_GIT_PARAMS no longer exists in latest version of husky

**Review by nicolasbrugneaux — APPROVED** [review]
